api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMediaDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.dao.impl;
+
+import javax.annotation.Nonnull;
+
+import java.util.List;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hibernate.Criteria;
+import org.openmrs.Obs;
+import org.openmrs.api.APIException;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.dao.FhirMediaDao;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.obs.ComplexObsHandler;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class FhirMediaDaoImpl extends BaseFhirDao<Obs> implements FhirMediaDao, ComplexObsHandler {
```

Comment:
This class definitely shouldn't be a `ComplexObsHandler`. That interface is only for classes which implement a specific type of `ComplexObs`, e.g., `ImageHandler` or `PdfHandler` or something like that.